server/rooms.ts

```diff
@@ -469,6 +469,7 @@ export abstract class BasicRoom {
 		Rooms.global.writeChatRoomData();
 	}
 	checkModjoin(user: User) {
+		if (user.hasSysopAccess()) return true;
```

Comment:
In fact, imagine if the entirety of `checkModjoin` were moved inside `Auth#atLeast` (so it would support `trusted` and `autoconfirmed` as possible values that could be passed to `atLeast`).

Reply:
Implemented.